fmn/formatters.py

```diff
@@ -0,0 +1,426 @@
+        email_message.add_header('X-Fedmsg-Topic', to_bytes(topic))
+    for category in categories or []:
+        email_message.add_header('X-Fedmsg-Category', to_bytes(category))
+    for username in usernames or []:
+        email_message.add_header('X-Fedmsg-Username', to_bytes(username))
+    for package in packages or []:
+        email_message.add_header('X-Fedmsg-Package', to_bytes(package))
+
+    subject_prefix = config.app_conf.get('fmn.email.subject_prefix', '')
+    if subject_prefix:
+        subject = '{0} {1}'.format(
+            subject_prefix.strip(), subject.strip())
+
+    email_message.add_header('Subject', to_bytes(subject))
+
+    # Since we do simple text email, adding the footer to the content
+    # before setting the payload.
+    footer = to_unicode(config.app_conf.get('fmn.email.footer', ''))
+
+    triggered_by = recipient['triggered_by_links']
+    if 'filter_id' in recipient and 'user' in recipient and triggered_by:
+        base_url = config.app_conf['fmn.base_url']
+        footer = (u'You received this message due to your preference settings at \n{base_url}'
+                  u'{user}/email/{filter_id}').format(base_url=base_url, **recipient) + footer
+    if footer:
+        content += u'\n\n--\n{0}'.format(footer.strip())
+
+    email_message.set_payload(to_bytes(content), 'utf-8')
+
+    # Explicitly declare encoding, but remove the transfer encoding
+    # https://github.com/fedora-infra/fmn/issues/94
+    email_message.set_charset('utf-8')
+
+    return email_message.as_string()
+
+
+def email_batch(messages, recipient):
+
+    if len(messages) == 1:
+        return email(messages[0], recipient)
+
+    def _format_line(msg):
+        timestamp = datetime.datetime.fromtimestamp(msg['timestamp'])
+        link = fedmsg.meta.msg2link(msg, **config.app_conf) or u''
+        payload = fedmsg.meta.msg2subtitle(msg, **config.app_conf) or u''
+
+        if recipient.get('verbose', True):
+            longform = fedmsg.meta.msg2long_form(msg, **config.app_conf) or u''
+            if longform:
+                payload += "\n" + longform
+
+        return timestamp.strftime("%c") + ", " + payload + "\n\t" + link
+
+    from_address = config.app_conf['fmn.email.from_address']
+    topics = set([message['topic'] for message in messages])
+    categories = set([topic.split('.')[3] for topic in topics])
+
+    def squash(items):
+        return reduce(set.union, items, set())
+    usernames = squash([
+        fedmsg.meta.msg2usernames(msg, **config.app_conf)
+        for msg in messages])
+    packages = squash([
+        fedmsg.meta.msg2packages(msg, **config.app_conf)
+        for msg in messages])
+    subject = u'Fedora Notifications Digest ({n} updates)'.format(n=len(messages))
+    summary = u"Digest summary:\n"
+    i = 0
+    for msg in messages:
+        i = i + 1
+        line = fedmsg.meta.msg2subtitle(msg, **config.app_conf) or u''
+        summary += '%d.\t%s\n' % (i, line)
```

Comment:
Why not use enumerate here instead of manually keeping count? You can do this instead:

```
for i, msg in enumerate(messages):
    line = fedmsg.meta.msg2subtitle(msg, **config.app_conf) or u''
    summary += '%d.\t%s\n' % (i + 1, line)
```